File: example/src/main/java/eu/darken/mvpbakery/example/screens/text/TextComponent.java
package eu.darken.mvpbakery.example.screens.text;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import dagger.Subcomponent;
import eu.darken.mvpbakery.injection.PresenterComponent;
import eu.darken.mvpbakery.injection.fragment.FragmentComponent;

@TextComponent.Scope
@Subcomponent()
public interface TextComponent extends FragmentComponent<TextFragment>, PresenterComponent<TextPresenter.View, TextPresenter> {

    @Subcomponent.Builder
    abstract class Builder extends FragmentComponent.Builder<TextFragment, TextComponent> {

    }

    @javax.inject.Scope
    @Retention(RetentionPolicy.RUNTIME)
    @interface Scope {
    }
}

File: example/src/main/java/eu/darken/mvpbakery/example/screens/text/TextComponent.kt
package eu.darken.mvpbakery.example.screens.text

import dagger.Subcomponent
import eu.darken.mvpbakery.injection.PresenterComponent
import eu.darken.mvpbakery.injection.fragment.FragmentComponent

@TextComponent.Scope
@Subcomponent
interface TextComponent : FragmentComponent<TextFragment>, PresenterComponent<TextPresenter, TextComponent> {

    @Subcomponent.Builder
    abstract class Builder : FragmentComponent.Builder<TextFragment, TextComponent>()

    @javax.inject.Scope
    @Retention(AnnotationRetention.RUNTIME)
    annotation class Scope
}
